src/utils.ts

```diff
@@ -312,3 +313,8 @@ export default abstract class BaseCommand extends Command {
     }
   }
 }
+
+export async function getPeerIdByAgreement (agreementReference: string): Promise<string | undefined> {
```

Comment:
Hmm, I think we should also prune the used PeerIds as I think they are one-time only usable information. Hence maybe after returning the PeerId here, I would delete the record from the DB? What do you think?

Reply:
Yea, good idea